File: widgets/ForwardWidgets-2kuai/scripts/validate-generated-data.py
import json
import sys

from data_contract import validate_data


if len(sys.argv) != 2:
    raise RuntimeError("Usage: python scripts/validate-generated-data.py <data-file>")

with open(sys.argv[1], "r", encoding="utf-8") as source:
    data = json.load(source)

record_count = validate_data(data)
print(f"[data] {sys.argv[1]}: {record_count} media records match the shared contract.")

File: widgets/ForwardWidgets-2kuai/scripts/validate-generated-data.js
const fs = require('node:fs');
const path = require('node:path');
const { validateData } = require('./lib/data-contract.cjs');

const filePath = process.argv[2];
if (!filePath) {
  throw new Error('Usage: node scripts/validate-generated-data.js <data-file>');
}

const data = JSON.parse(fs.readFileSync(filePath, 'utf8'));
const recordCount = validateData(data);
console.log(`[data] ${path.basename(filePath)}: ${recordCount} media records match the shared contract.`);
